[do-not-merge] Add initial operator quickstarts for k8s, and openshift

rmarting opened the pull request:
Basic examples to deploy the operator, and operate a keycloak instance in the common environments: k8s, ocp

Closes #492

mposolda commented:
@rmarting There should not be PR sent to the branch `latest` . All PRs to quickstarts should go to branch `main` .

The content of branch `latest` is always overwritten from branch `main` during Keycloak release, so your changes to the branch `latest` would be lost if updates are done only in branch `latest` instead of updating branch `main` . We plan to update quickstarts contributon guide with mentioning this.

rmarting commented:
Thanks @mposolda! Could you check that now the PR is in a better position to move forward? I changed it to the `main` branch and resolved some of the conflicts.
HTH

rmarting commented:
This PR is closed and moved the content to the #563 PR, created from the `main` branch and added the signed commit.

@mposolda Please, keep in mind the new PR, instead of this original one. The PR is clean instead of the multiple commits here because it was started from the `latest` branch. HTH
